src/utils.py

```diff
@@ -429,6 +429,37 @@ def _run_heuristic_search(
 
     If no goal is found, returns the state with the best priority.
     """
+    cur_node: _HeuristicSearchNode[_S, _A] = _HeuristicSearchNode(
```

Comment:
can you make this its own search function? it seems generally useful. is it hill climbing?